src/Octoshift/AdoClient.cs

```diff
@@ -15,7 +15,7 @@ public class AdoClient
         private readonly OctoLogger _log;
         private double _retryDelay;
 
-        public AdoClient(OctoLogger log, HttpClient httpClient, string personalAccessToken)
+        public AdoClient(OctoLogger log, HttpClient httpClient, VersionChecker versionChecker, string personalAccessToken)
```

Comment:
Instead of introducing a new dependency just to get the assembly's version, we can just create an extension method for `HttpRequestHeaders` and get rid of the unnecessary dependency:
```c#
public static void AddUserAgent(this HttpRequestHeaders httpRequestHeaders)
{
    httpRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("OctoshiftCLI", Assembly.GetExecutingAssembly().GetName().Version.ToString()));
}
```

Reply:
I'm not sure I agree. Having it encapsulated in the VersionChecker centralizes a few things around how we deal with version numbers. For example is it the 4-part version number, or 2 part like in the GH releases? Does the version number include the "v" before it or not? I tried to keep all those details hidden away in VersionChecker so other classes wouldn't need to worry about it.

Reply:
Ok, but in that case I would create an interface `IVersionProvider` which has that one method in it, the implementation would still be inside the `VersionChecker` but then we would only inject `IVersionProvider` instead of the entire `VersionChecker`. This will help making future refactors, etc. easier since we don't depend on functionality that we are not ever going to need.